### PR TITLE
feat: add pagination

### PR DESCRIPTION
**Pull request by tom1145**
Fixes [#22](https://github.com/oceanprotocol/nodes-dashboard/issues/22).

Changes proposed in this PR:

- added pagination to table

### CODE REVIEW

**Review by AdriGeorge — APPROVED** [review]

**Comment by bogdanfazakas** [comment]
I have identified 2 thinks for the moment see points bellow:

1. Before merging this we need to also update the logic and create a new context for the map & locations, because it displays only the nodes 
Screenshot 2024-09-12 at 10 50 59
on the map received for the given page

2. the results seem to be capped at 10k, but we have more than this number of nodes, but this may require changes to the incentive-backend
Screenshot 2024-09-12 at 10 50 59

**Comment by bogdanfazakas** [comment]
> I have identified 2 thinks for the moment see points bellow:
> 
> 1. Before merging this we need to also update the logic and create a new context for the map & locations, because it displays only the nodes  on the map received for the given page

> 2. the results seem to be capped at 10k, but we have more than this number of nodes, but this may require changes to the incentive-backend
> 
> 

think it we can also move fwd if we fix point 2 and address 1 with map & locations on a later pr

**Comment by tom1145** [comment]
> > I have identified 2 thinks for the moment see points bellow:
> > 
> > 1. Before merging this we need to also update the logic and create a new context for the map & locations, because it displays only the nodes  on the map received for the given page
> 
> > 2. the results seem to be capped at 10k, but we have more than this number of nodes, but this may require changes to the incentive-backend
> 
> think it we can also move fwd if we fix point 2 and address 1 with map & locations on a later pr

I've made a fix now with totalItems and sort for uptime, can you please check it out? thanks!

File: src/components/Toolbar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  onReset
+}) => {
+  return (
+    <GridToolbarContainer>
+      <GridToolbarColumnsButton />
+      <GridToolbarFilterButton />
+      <GridToolbarDensitySelector />
+      <GridToolbarExport />
+
+      <TextField
```
Comment (bogdanfazakas) [inline src/components/Toolbar/index.tsx:34]: i would align all this elements related to search to the headers right , it should work with some `align-items: flex-end`  i suppose

**Pull request merged**